sysdump/sysdump.go

```diff
@@ -687,6 +687,32 @@ func (c *Collector) Run() error {
 	return nil
 }
 
+func (c *Collector) createOutputFile(d string, path func(string) string) (string, error) {
+	// Create the zip file.
+	f := path(outputZipFilename)
+	if err := archiver.Archive([]string{d}, f); err != nil {
+		return "", fmt.Errorf("failed to create zip file: %w", err)
+	}
+	// Encrypt it if asked to.
+	if c.options.Encrypt {
+		e, err := createEncryptedZipFile(f, c.options.EncryptionKey)
+		if err != nil {
+			return "", fmt.Errorf("failed to encrypt zip file: %w", err)
+		}
+		f = e
+	}
+	// Move the output file to the current directory.
+	p, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("failed to get current directory: %w", err)
+	}
+	r := filepath.Join(p, filepath.Base(f))
+	if err := os.Rename(f, r); err != nil {
```

Comment:
Note: This fails on my machine with a `LinkError` due to `/tmp` being on a different filesystem (which I think is quite common on linux).

I would propose creating a temporary file next to the target file (e.g. via `os.CreateTemp`) and then use `os.Rename` instead. It's probably also better not to put potentially sensitive data in `/tmp`

Reply:
Hmmm that's interesting 🤔 I'll try to have a look.

Reply:
Fixed!